tech/tech.py

```diff
@@ -163,92 +166,134 @@ def impact_hours_rewards(self):
+            impact_hour_rate = R* (self.minimum_raise / (self.minimum_raise + m*H))
+            cultural_build_tribute = (H * impact_hour_rate)/self.minimum_raise
+            df_hatch_params = df_hatch_params.append({'Total XDAI Raised': self.minimum_raise, 'Impact Hour Rate':impact_hour_rate, 'Cultural Build Tribute':cultural_build_tribute, 'Hatch tribute':hatch_tribute, 'Redeemable':hatch_tribute/(1 + cultural_build_tribute), 'label':'Min Raise'}, ignore_index=True)
+            df_hatch_params = df_hatch_params.sort_values(['Total XDAI Raised'])
+
+        df_min_raise = df_hatch_params.query("label == 'Min Raise'")
+        if len(df_min_raise) > 1:
+            df_hatch_params = df_hatch_params.drop(df_min_raise.first_valid_index())
+
+        # Add 'Tarfet Raise' label case there is already a row with target_raise value
+        df_hatch_params.loc[df_hatch_params['Total XDAI Raised'] == self.target_raise, 'label'] = "Target Raise"
+
+        # Add a new row with target_raise vale case there is no row with its value
+        if "Target Raise" not in df_hatch_params['label']:
+            impact_hour_rate = R* (self.target_raise / (self.target_raise + m*H))
+            cultural_build_tribute = (H * impact_hour_rate)/self.target_raise
+            df_hatch_params = df_hatch_params.append({'Total XDAI Raised': self.target_raise, 'Impact Hour Rate':impact_hour_rate, 'Cultural Build Tribute':cultural_build_tribute, 'Hatch tribute':hatch_tribute, 'Redeemable':hatch_tribute/(1 + cultural_build_tribute), 'label':'Target Raise'}, ignore_index=True)
+            df_hatch_params = df_hatch_params.sort_values(['Total XDAI Raised'])
+
+        df_target_raise = df_hatch_params.query("label == 'Target Raise'")
+        if len(df_target_raise) > 1:
+            df_hatch_params = df_hatch_params.drop(df_target_raise.first_valid_index())
+
+        # Add 'Max Raise' label case there is already a row with max_raise value
+        df_hatch_params.loc[df_hatch_params['Total XDAI Raised'] == self.maximum_raise, 'label'] = "Max Raise"
+
+        # Add a new row with max_raise vale case there is no row with its value
+        if "Max Raise" not in df_hatch_params['label']:
+            impact_hour_rate = R* (self.maximum_raise / (self.maximum_raise + m*H))
+            cultural_build_tribute = (H * impact_hour_rate)/self.maximum_raise
+            df_hatch_params = df_hatch_params.append({'Total XDAI Raised': self.maximum_raise, 'Impact Hour Rate':impact_hour_rate, 'Cultural Build Tribute':cultural_build_tribute, 'Hatch tribute':hatch_tribute, 'Redeemable':hatch_tribute/(1 + cultural_build_tribute), 'label':'Max Raise'}, ignore_index=True)
+            df_hatch_params = df_hatch_params.sort_values(['Total XDAI Raised'])
+
+        df_max_raise = df_hatch_params.query("label == 'Max Raise'")
+        if len(df_max_raise) > 1:
+            df_hatch_params = df_hatch_params.drop(df_target_raise.first_valid_index())
```

Comment:
```suggestion
            df_hatch_params = df_hatch_params.drop(df_max_raise.first_valid_index())
```
 
I'm not sure if it's wrong AHAHAH

Reply:
Yeah. It was wrong, thanks for finding it!